Add sample which creates a new VM from vmdk file

atodorov opened the pull request:
I need help with this sample.

Around line 100 I already have the VM created and I want to attach an existing vmdk file as its IDE 0 disk. I'm not quite sure which API calls I need and in what order.

atodorov committed:
Add sample which creates a new VM from vmdk file

prziborowski commented on samples/create_vm.py:
```diff
@@ -0,0 +1,184 @@
+
+    disk_spec = vim.vm.device.VirtualDeviceSpec(
+        device=device,
+        fileOperation="create",
```
You should remove this line as it indicates to create a new file.
If it already exists then you don't need a fileOperation.

prziborowski commented on samples/create_vm.py:
```diff
@@ -0,0 +1,184 @@
+
+    ### good till here
+    device=vim.vm.device.VirtualDisk(
+        controllerKey=None,
```
You will want a controllerKey that points to some controller.
If you said you want ide, then it should either be an existing IDEController, or you will need to add one.
If it is an existing one, use that device.key.
If you are adding a new one, set the device.key to a negative number and use that same number for the controllerKey here.

atodorov commented:
@prziborowski thanks for your tips. As of now I am able to create a VM and later attach the vmdk file as it's IDE 0 disk. I've inspected the resulting configuration and it looks similar to what I get when I create a VM manually (minus the attached networking interface). 

My problem ATM is that the VM doesn't boot. If I try to power it on I get the following message in vCenter UI:

Module 'DevicePowerOn' power on failed

I have no idea what this is and how do I fix it. Suggestions are welcome. Meanwhile I will be searching for solution.

prziborowski commented on samples/create_vm.py:
```diff
@@ -0,0 +1,210 @@
+
+    disk_spec = vim.vm.device.VirtualDeviceSpec(
+        device=vim.vm.device.VirtualDisk(
+            unitNumber=1,
```
I tried the script and the error I received (I had to look at the vmware.log file of the VM) was:
2018-10-31T17:56:52.615Z| vmx| I125+ The IDE device (disks/CD-ROM) configuration is incorrect. There is an IDE slave with no master at 'ide0:1'. This configuration does not work correctly in virtual machines. Move the disk/CD-ROM from 'ide0:1' to 'ide0:0' using the configuration editor.

If you set this to unitNumber=0, then I could power on.
(Alternatively you could attach another IDE device in slot 0 as the master).

atodorov replied:
Thank you, power on works now.

atodorov commented:
@prziborowski this PR should be ready for review/merge. Thanks for your tips.

atodorov committed:
Re-use tools.cli.build_arg_parser() and handle disabled SSL verification

atodorov committed:
Display domain name and IP address

atodorov commented:
Also updated `find_by_uuid.py` to allow skipping of SSL verification and printing IP address and domain name of the VM guest. Please review and merge.

atodorov committed:
Refactor scheduled_poweroff

- allow to disable SSL verification
- re-use cli.build_arg_parser()
- make --date not mandatory and execute power off immediately
  if value is not set

atodorov committed:
Update SSL verify and VM search for destroy_vm.py
